### ISSUE TITLE
Special From Date set to today's date when Use Default Date checked in Store scope

### ISSUE THREAD

**Issue by squeegy06** [magento/magento2#22511]
### Preconditions (*)

1. Magento 2.2.8
2. Configured product with Special Price, Special From Date, and Special To Date set at the All Stores Scope.

### Steps to reproduce (*)

1. Edit a product in store view (say Default Store View) that has a Special Price, Special From Date, and Special To Date already set at the All Store Scope.
2. Open the Advanced Pricing popup.
3. Note that the Use Default Value boxes should be checked for Special From Date and Special To Date, if not, check them and close the Advance Price Window.
4. Save the product.
5. After the page reloads, open the Advanced Price window again.
6. Observe Screen.

### Expected result (*)
I expected the Use Default Value boxes to remain checked on both the Special From Date and Special To Date and for them to display the values set at the All Store Scope.

### Actual result (*)
The Special From Date "Use Default Value" is unchecked and the value is set to today's date.

While debugging this I found `Magento\Catalog\Observer\SetSpecialPriceStartDate` which is set to run before product save.  It has the following function

```
    /**
     * Set the current date to Special Price From attribute if it empty
     *
     * @param \Magento\Framework\Event\Observer $observer
     * @return $this
     */
    public function execute(\Magento\Framework\Event\Observer $observer)
    {
        /** @var  $product \Magento\Catalog\Model\Product */
        $product = $observer->getEvent()->getProduct();
        if ($product->getSpecialPrice() && !$product->getSpecialFromDate()) {
            $product->setData('special_from_date', $this->localeDate->date());
        }

        return $this;
    }
```

Which appears to be setting the Special From Date to today if there's no Special From Date set, but a Special Price is set.  Earlier in the execution of the save though, `Magento\Catalog\Controller\Adminhtml\Product\Initialization\Helper` set's the values of attributes that are set to "Use Default Value" to either `null` or `false` depending on the type of attribute.  So when this observer is ran, it get's `null` for the Special From Date, and since we have a Special From Price set, it ignores the "Use Default Value" and sets the Special From Date.

I'm not entirely sure though what to do for a solution to this.  For the moment though I've removed the Observer to prevent the undesired behavior.  But it seems to me like this should not be the behavior when saving products at the store level.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @AlexWorking 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99435` were created

**Issue Available**: @AlexWorking, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @squeegy06. Thank you for your report. 
The issue has been fixed in magento/magento2#22521 by @maheshWebkul721 in 2.3-develop branch
Related commit(s):
 - [db71c65a247ccc78bce17e0f4bc44f1300e67c02](https://github.com/magento/magento2/commit/db71c65a247ccc78bce17e0f4bc44f1300e67c02)

The fix will be available with the upcoming 2.3.3 release.
